### PR TITLE
[expo-go] Improve error messaging with snack

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes ENG-12176
Currently when running an incompatible snack in expo go the error messages do not differentiate between a snack and a normal project.

# How
Added a check for snack URLs and provided different messaging based on this.

# Test Plan
Tested with compatible and incompatible snacks also did the same with projects. Screen behaves correctly for each case.

| iOS | Android |
| -------- | ------- |
|iosSnack| androidSnack
|

### CODE REVIEW

**Review by wschurman — APPROVED** [review]
Will defer to @byCedric for message wording review but code looks good.

**Review by gabrieldonadel — APPROVED** [review]
LGTM

File: apps/expo-go/ios/Exponent/Kernel/AppLoader/CachedResource/EXManifestResource.m [hunk 1]
```diff
@@ -326,35 +325,42 @@ - (NSError *)formatError:(NSError *)error
...
-
     showTryAgainButton = false;
     formattedMessage = [NSString stringWithFormat:
-                          @"• The installed version of Expo Go is for **%@ %@**.\n"
-                        @"• The project you opened uses **SDK %d**.",
-                        maybePluralSdkString,
+                          @"• The installed version of Expo Go is for **SDK %@**.\n"
+                        @"• The %@ you opened uses **SDK %d**.",
                         [self supportedSdkVersionsConjunctionString:@"and"],
+                        self.isSnackURL ? @"snack" : @"project",
```
Comment (brentvatne) [inline apps/expo-go/ios/Exponent/Kernel/AppLoader/CachedResource/EXManifestResource.m:333]:
```suggestion
                        self.isSnackURL ? @"Snack" : @"project",
```

File: apps/expo-go/ios/Exponent/Kernel/AppLoader/CachedResource/EXManifestResource.m [hunk 2]
```diff
@@ -326,35 +325,42 @@ - (NSError *)formatError:(NSError *)error
...
                         requiredVersionNum
     ];
 
 #if (TARGET_OS_SIMULATOR)
-    fixInstructions = [NSString stringWithFormat:@"Either upgrade this project to SDK %@ or install a version of Expo Go that is compatible with your project.\n\n"
+    fixInstructions = [NSString stringWithFormat:@"Either upgrade this %@ to SDK %@ or install a version of Expo Go that is compatible with your project.\n\n"
                        @"[https://docs.expo.dev/workflow/upgrading-expo-sdk-walkthrough/](Learn how to upgrade to SDK %d.)\n\n"
                        @"[%@](Learn how to install Expo Go for SDK %d.)",
+                       self.isSnackURL ? @"snack" : @"project",
```
Comment (brentvatne) [inline apps/expo-go/ios/Exponent/Kernel/AppLoader/CachedResource/EXManifestResource.m:341]:
```suggestion
                       self.isSnackURL ? @"Snack" : @"project",
```

**Review by brentvatne — APPROVED** [review]
let's get rid of the "Learn how to upgrade to..." button since that is specific to upgrading projects and not Snacks, and we already explain how to upgrade a Snack on this screen (selecting the version)

**Pull request merged**